bib/cmd/bootc-image-builder/image_test.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/osbuild/bootc-image-builder/bib/internal/source"
+	"github.com/osbuild/images/pkg/manifest"
+	"github.com/osbuild/images/pkg/runner"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetDistroAndRunner(t *testing.T) {
+	cases := []struct {
+		id             string
+		versionID      string
+		expectedDistro manifest.Distro
+		expectedRunner runner.Runner
+		expectedErr    string
+	}{
+		// Happy
+		{"fedora", "40", manifest.DISTRO_FEDORA, &runner.Fedora{Version: 40}, ""},
+		{"centos", "9", manifest.DISTRO_EL9, &runner.CentOS{Version: 9}, ""},
+		{"centos", "10", manifest.DISTRO_EL10, &runner.CentOS{Version: 10}, ""},
+		{"centos", "11", manifest.DISTRO_NULL, &runner.CentOS{Version: 11}, ""},
+		{"rhel", "9.4", manifest.DISTRO_EL9, &runner.RHEL{Major: 9, Minor: 4}, ""},
+		{"rhel", "10.4", manifest.DISTRO_EL10, &runner.RHEL{Major: 10, Minor: 4}, ""},
+		{"rhel", "11.4", manifest.DISTRO_NULL, &runner.RHEL{Major: 11, Minor: 4}, ""},
+		{"toucanos", "42", manifest.DISTRO_NULL, &runner.Linux{}, ""},
+
+		// Sad
+		{"fedora", "asdf", manifest.DISTRO_NULL, nil, "cannot parse Fedora version (asdf)"},
+		{"centos", "asdf", manifest.DISTRO_NULL, nil, "cannot parse CentOS version (asdf)"},
+		{"rhel", "10", manifest.DISTRO_NULL, nil, "invalid RHEL version format: 10"},
+		{"rhel", "10.asdf", manifest.DISTRO_NULL, nil, "cannot parse RHEL minor version (asdf)"},
+	}
+
+	for _, c := range cases {
+		t.Run(fmt.Sprintf("%s-%s", c.id, c.versionID), func(t *testing.T) {
+			osRelease := source.OSRelease{
+				ID:        c.id,
+				VersionID: c.versionID,
+			}
+			distro, runner, err := getDistroAndRunner(osRelease)
+			if c.expectedErr != "" {
+				assert.ErrorContains(t, err, c.expectedErr)
```

Comment:
(super nitpick) might be worthwhile to test to spell the full error out here to see how the full thing looks like, i.e. if there is (too much) duplication from wrapping the strconv error (but super minor)